realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1002,4 +1003,39 @@ public void run() {
             }
         });
     }
+
+    @Test
+    @RunTestInLooperThread
+    public void deleteAndDeleteAll() {
+        final Realm realm = looperThread.realm;
+        realm.beginTransaction();
+        for (int i = 0; i < 10; i++) {
+            StringOnly stringOnly = realm.createObject(StringOnly.class);
+            stringOnly.setChars("String " + i);
+        }
+        realm.commitTransaction();
+
+        RealmResults<StringOnly> stringOnlies = realm.where(StringOnly.class).findAll();
+
+        realm.beginTransaction();
+        // remove one object
+        stringOnlies.get(0).deleteFromRealm();
+        realm.commitTransaction();
+
+        realm.beginTransaction();
+        // remove the rest
+        stringOnlies.deleteAllFromRealm();
+        realm.commitTransaction();
+
+        assertEquals(0, realm.where(StringOnly.class).findAll().size());
+
+        // The above commit should have put a REALM_CHANGED event on the Looper queue before this runnable.
+        looperThread.postRunnable(new Runnable() {
```

Comment:
No need for this or the `@RunTestInLooperTread`. Just ending the test with the assert should be fine.